param/parameterized.py

```diff
@@ -1985,6 +1990,8 @@ def _setup_refs(self_, refs):
             ))
 
     def _update_ref(self_, name, ref):
+        if name in self_.self._param__private.async_refs:
+            self_.self._param__private.async_refs.pop(name).cancel()
         for _, watcher in self_.self._param__private.ref_watchers:
```

Comment:
```suggestion
        param_private = self_.self._param__private
        if name in param_private.async_refs:
            param_private.async_refs.pop(name).cancel()
        for _, watcher in param_private.ref_watchers:
```

It is mainly for readability, but it is more of a personal preference, so you can ignore it.